### PR TITLE
Make the default fluent.conf a confd template

### PR DESCRIPTION
**Pull request by ghost**
## Motivation

I would like to determine the setting of `fluentd` when running the `kube-fluentd` image.
So I made `fluent.conf` supported to `confd`.

## Attention

Only the minimal template variable of `confd` in `fluent.conf`.

### CODE REVIEW

File: Dockerfile.template [hunk 1]
```diff
@@ -64,10 +64,6 @@ WORKDIR /
 
 COPY rootfs /
 
-COPY fluent.conf /fluentd/etc/
-
-RUN sed -i -e 's/type/@type/g' /fluentd/etc/fluent.conf
```
Comment (ghost) [inline Dockerfile.template:69]: Drop support of fluentd v0.12 or less.

File: rootfs/etc/confd/templates/fluent.conf.tmpl [hunk 2]
```diff
@@ -193,22 +193,22 @@
   buffer_queue_full_action drop_oldest_chunk
   # Set the chunk limit conservatively to avoid exceeding the GCL limit
   # of 10MiB per write request.
-  buffer_chunk_limit 2M
+  buffer_chunk_limit {{getv "/fluentd/out/kubesys/buffer/chunk/limit"}}
```
Comment (mumoshu) [inline rootfs/etc/confd/templates/fluent.conf.tmpl:196]: Could we move the default settings like `2M` here to `ENV` steps of our Dockerfile?
```
ENV FLUENTD_OUT_KUBESYS_BUFFER_CHUNK_LIMIT=2M
...
```

File: Dockerfile.template (hunk 1 above)
Comment (mumoshu) [reply Dockerfile.template:69]: That's the way to go. I'm ok with dropping it.

**Comment by mumoshu** [comment]
This looks super handy!
Thanks for your contribution 👍

File: rootfs/etc/confd/templates/fluent.conf.tmpl (hunk 2 above)
Comment (ghost) [reply rootfs/etc/confd/templates/fluent.conf.tmpl:196]: Nice idea!
I'll do it.

Comment (ghost) [reply rootfs/etc/confd/templates/fluent.conf.tmpl:196]: @mumoshu 
I pushed.
Please check it.

**Comment by mumoshu** [comment]
LGTM. Thanks again for your contribution Hayashi-san!

**Pull request merged**